internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -141,6 +215,16 @@ func (r MongoClusterResource) Arguments() map[string]*pluginsdk.Schema {
 			RequiredWith: []string{"administrator_username"},
 		},
 
+		"auth_config_allowed_modes": {
+			Type:     pluginsdk.TypeSet,
+			Optional: true,
+			Computed: true,
```

Comment:
Missing required O+C comment between Optional and Computed properties. Add a comment explaining why this property is Optional+Computed.

Reference: [Best Practices](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/topics/best-practices.md#setting-properties-to-optional--computed)